lib/insights/api/common/audit_log.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require_relative 'audit_log/formatter'
+require_relative 'audit_log/middleware'
+
+module Insights::API::Common
+  # Audit Logger into selected logger, but primarily to CloudWatch
+  class AuditLog
+    class << self
+      def logger
+        @logger ||= init_logger
+      end
+
+      def logger=(logger)
+        @logger = init_logger(logger)
+      end
+
+      def setup(logger = nil)
+        self.logger = logger
+      end
+
+      def with_account(account_number)
+        original = Thread.current[:audit_account_number]
+        Thread.current[:audit_account_number] = account_number
+        return unless block_given?
+
+        yield
+        Thread.current[:audit_account_number] = original
```

Comment:
We use hooks for the auth, which means that the `account_number` would be only setting. Ensuring that the account num is reset is one thing I'm working on. It would not be a 100%, but at least it should cover controllers.

Reply:
I suggest setting the original account number in `ensure` block in case of exception